Fixes HA WARNING message during startup. Defines default_entity_id based on platform and object_id. Removes deprecated object_id

IgnacioHR opened the pull request:
**Describe what the PR does:**

Fixes HA WARNING message during startup. Renames object_id with default_entity_id

**Does this fix a specific issue?**

Fixes https://github.com/bachya/ecowitt2mqtt/issues/1324

No new functionality added!

**Checklist:**

- [x] Confirm that one or more new tests are written for the new functionality.
- [x] Run tests and ensure everything passes (with 100% test coverage).
- [x] Update `README.md` with any new documentation.

IgnacioHR committed:
Rename object_id to default_entity_id in hass.py

Fixes MQTT integration issue at HA startup

IgnacioHR committed:
Merge pull request #1 from IgnacioHR/IgnacioHR-patch-1

Rename object_id to default_entity_id in hass.py

andrew-codechimp commented:
According to the log the default_entity_id should include the platform before the prefix. I don't think just renaming the key is enough. 

"default_entity_id": "sensor.E2m_soilad7"

IgnacioHR commented:
Hi,

Thanks for reviewing the PR.

I wanted to clarify a point regarding your comment that “the default_entity_id should include the platform before the prefix, and that simply renaming the key is not enough.” In this context, my understanding of “platform” matches what Home Assistant refers to as the domain (sensor, binary_sensor, etc.).

The Home Assistant warning explicitly states that the new value should be the full entity_id, including the domain, for example:

`"default_entity_id": "sensor.E2m_soilad7"
`

In other words, it expects exactly the old `object_id` prefixed with the domain. This aligns with the standard MQTT discovery format, where the entity ID is constructed as `<platform>.<object_id>`.

Of course, if in the future you plan to redesign how identifiers are generated or restructure the discovery payloads more broadly, that would be great — but this minimal change already resolves the incompatibility and allows the integration to move forward without warnings.

Thanks again for your work on the project.

andrew-codechimp commented:
Hi, firstly I'm not the owner of this project, just the person that raised the issue you linked to.

You are right, Platform is an internal name within HA for what are base domains so sensor/switch etc.

I don't think your code change adds the sensor part, I did look at this and there are a couple different domains we'd have to pass through to this function to do the prefix.

IgnacioHR commented:
> Hi, firstly I'm not the owner of this project, just the person that raised the issue you linked to.
> 
> You are right, Platform is an internal name within HA for what are base domains so sensor/switch etc.
> 
> I don't think your code change adds the sensor part, I did look at this and there are a couple different domains we'd have to pass through to this function to do the prefix.

Digging a little bit into the MQTT server I think you are right, I'm going to change the PR

<img width="1232" height="701" alt="image" src="https://github.com/user-attachments/assets/b4f7a4e5-5ac6-41c2-be7f-c7423e05f4a1" />

IgnacioHR committed:
Add default_entity_id to MQTT discovery configuration

rolled back previous change to keep object_id
added default_entity_id using platform.object_id

andrew-codechimp commented on ecowitt2mqtt/helpers/publisher/mqtt/hass.py:
```diff
@@ -519,6 +520,7 @@ def _get_discovery_info(
 
         if self._config.hass_entity_id_prefix:
             discovery.object_id = f"{self._config.hass_entity_id_prefix}_{payload_key}"
```
You shouldn't be sending object_id, which is deprecated.  Suggest moving this f string to within the  default_entity_id one.

IgnacioHR replied:
Done,

Hope HA updates the [documentation](https://www.home-assistant.io/integrations/mqtt/#discovery-messages) as it still refers to the object_id

IgnacioHR committed:
Refactor entity ID handling in hass.py

Removed object_id assignment and updated default_entity_id construction to include hass_entity_id_prefix.

andrew-codechimp commented on ecowitt2mqtt/helpers/publisher/mqtt/hass.py:
```diff
@@ -1,4 +1,4 @@
-"""Define MQTT publishing."""
+ƒ"""Define MQTT publishing."""
```
The ƒ shouldn't be there and making tests fail.
